mne/io/snirf/_snirf.py

```diff
@@ -0,0 +1,202 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#
+# License: BSD (3-clause)
+
+import re as re
+import numpy as np
+
+from ..base import BaseRaw
+from ..meas_info import create_info
+from ...annotations import Annotations
+from ...utils import logger, verbose, fill_doc, warn, requires_h5py
+
+
+@requires_h5py
```

Comment:
This is really a testing decorator not meant to be used by user-facing functions. Usually we do this:
```
if not check_version('h5py'):
    raise ImportError('The h5py package is required to read raw SNIRF data')
```
If you want, you could add a helper to `mne/utils/check.py` to make this process more general, like:
```
def _require_version(lib, what, version='0.0'):
    if not check_version(lib, version):
        extra = f' (version >= {version})' if version != '0.0' else ''
        raise ImportError(f'The {lib} package{extra} is required to {what}')
```
WDYT?

Reply:
Sounds good. Ive added this as you suggest. I have not gone through the code base to edit other functions. But will do that once this is merged as a seperate PR.

Reply:
I figure I can do a search for all uses of `check_version` and see which match this pattern. Any other suggestions?

Reply:
That's all I would do. Don't need to catch all of them, we can always migrate as we're working on bits of code anyway.